src/v/datalake/record_schema_resolver.cc

```diff
@@ -60,9 +64,46 @@ struct schema_translating_visitor {
         }
     }
     ss::future<checked<type_and_buf, record_schema_resolver::errc>>
-    operator()(const ppsr::protobuf_schema_definition&) {
-        // XXX: a subsequent PR will add protobuf support.
-        co_return type_and_buf::make_raw_binary(std::move(buf_no_id));
+    operator()(const ppsr::protobuf_schema_definition& pb_def) {
+        const google::protobuf::Descriptor* d;
+        proto_offsets_message_data offsets;
+        try {
+            auto offsets_res = get_proto_offsets(buf_no_id);
+            if (offsets_res.has_error()) {
+                co_return record_schema_resolver::errc::bad_input;
+            }
+            offsets = std::move(offsets_res.value());
+            // TODO: maybe there's another caching opportunity here.
+            auto d_res = descriptor(pb_def, offsets.protobuf_offsets);
+            if (d_res.has_error()) {
+                co_return record_schema_resolver::errc::bad_input;
+            }
+            d = &d_res.value().get();
+        } catch (...) {
```

Comment:
What is the point of using outcome if stuff can throw too? We really should not have so many ways to do things. It will cause a bug when we expect something to return an error code and it instead throws. And having to handle errors in two ways is tedious and verbose.

Reply:
yeh agree. if the entire stack is using error codes, then you can effectively have a top-level catch that is heavy handed.

Reply:
Fair, I think the exceptions thrown in get_proto_offsets() are an oversight. Things that return result types shouldn't throw. Will put up a followup